app/main/util/render_utils.py

```diff
@@ -56,37 +54,28 @@ def get_download_filename(file):
     return None
 
 
-def manage_static_file(file, file_extension):
-    """Manage the file in the static directory."""
+def create_presigned_url(file):
+    file_extension = file.FileName.split(".")[-1].lower()
+    if file_extension not in current_app.config["SUPPORTED_RENDER_EXTENSIONS"]:
+        current_app.logger.info(
+            f"Rendering file format '{file_extension}' is not currently supported by AYR."
+        )
+        return None
+
     s3 = boto3.client("s3")
     bucket = current_app.config["RECORD_BUCKET_NAME"]
     key = f"{file.consignment.ConsignmentReference}/{file.FileId}"
-    files_directory = os.path.join(current_app.static_folder, "files")
-    static_file_path = ""
-
-    if file_extension == "pdf":
-        static_file_path = os.path.join(files_directory, "temp_file.pdf")
-    elif file_extension in ["png", "jpg", "jpeg"]:
-        static_file_path = os.path.join(
-            files_directory, f"temp_file.{file_extension}"
-        )
-    else:
-        static_file_path = os.path.join(
-            files_directory, f"temp_file.{file_extension}"
-        )
-
-    if os.path.exists(files_directory):
-        shutil.rmtree(files_directory)
-
-    os.makedirs(files_directory)
 
-    s3_file_object = s3.get_object(Bucket=bucket, Key=key)
-    file_content = s3_file_object["Body"].read()
+    try:
+        presigned_url = s3.generate_presigned_url(
+            "get_object", Params={"Bucket": bucket, "Key": key}, ExpiresIn=10
+        )
 
-    with open(static_file_path, "wb") as static_file:
-        static_file.write(file_content)
+    except Exception as e:
+        current_app.logger.info(f"Failed to create presigned url. {e}")
```

Comment:
since we have the catch at the higher level, do we need this here? the messages seem to be pretty similar, so having 2 different messages may be confusing?

And if any of the rest of this function fails we currently error silently (with the exception at the higher level), whereas if `s3.generate_presigned_url(` errors, we abort  with 404

Reply:
Changed to only keep the try except at the routes.py level